sympy/functions/elementary/exponential.py

```diff
@@ -262,6 +262,12 @@ def eval(cls, arg):
                             return -S.ImaginaryUnit
                         elif ask(Q.odd(coeff + S.Half)):
                             return S.ImaginaryUnit
+                    elif coeff.is_Rational:
```

Comment:
```suggestion
                    elif coeff.is_rational:
```

This will allow symbolic coefficients to pass.

Reply:
Hi, thanks for looking through the changes! Wouldn't this proposed change entail that in case of a symbolic argument to exp(), another term (multiple of `2*pi*I`) would be added to the argument and hence it'd grow slightly longer and arguably more complex? Whereas for `is_Rational` the modulo-reduction would only be applied if it's an actual (albeit minor) simplification.
I may well be missing something though.

Reply:
I think what you have now will catch this case in the `(2*coeff).is_integer` block.